agent/src/beerocks/slave/agent_db.h

```diff
@@ -72,8 +75,165 @@ class AgentDB {
     void db_lock() { m_db_mutex.lock(); }
     void db_unlock() { m_db_mutex.unlock(); }
 
+    /* Put down from here database members and functions used by the Agent modules */
+
 public:
-    /* Put here database members used by the Agent modules */
+    /* Agent Configuration */
+    struct sDeviceConf {
+        struct sFrontRadio {
+
+        } front_radio;
+
+        struct sBackRadio {
+
+        } back_radio;
+
+        bool local_gw;
+        bool local_controller;
+    } device_conf;
+
+    /** 
+     * Agent Sub Entities Data
+     */
+    struct sBridge {
+        sMacAddr mac;
+        std::string iface_name;
+    } bridge;
+
+    struct sBackhaul {
+        enum class eConnectionType { Invalid = 0, Wired, Wireless } connection_type;
```

Comment:
It is totally ok to not split enum type definition and declaration.
Second, you don't see it being used now since it is a preparative commit. It will be used in one of the follow-up commits on this PR.